Mathlib/Data/ENNReal/Floor.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2024 Kalle Kytölä. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kalle Kytölä
+-/
+import Mathlib.Topology.Instances.ENNReal
+
+/-!
+# A floor function on the extended non-negative reals
+-/
+
+section ENNReal_floor
+
+open Filter BigOperators TopologicalSpace Topology Set ENNReal NNReal
+
+namespace ENNReal
+
+/-- The floor function `ℝ≥0∞ → ℝ≥0∞`: the floor of `x` is the supremum of the (coerced) natural
+numbers `n` satisfying `n ≤ x`. -/
+noncomputable def floor (x : ℝ≥0∞) : ℝ≥0∞ :=
```

Comment:
Should it take values in `ENat`?

Reply:
@YaelDillies made the same remark when I mentioned this. My eventual use case would coerce back to `ENNReal` anyway, but I think you have a good point and the floor function might be more naturally be defined as `ENat` valued.

I modified the PR accordingly, except I only made the necessary `ENat` topology instance very quickly in this file locally and `sorry`ed out the proof that it is the order topology. If this approach looks otherwise acceptable, I could see if I can fill in the necessary parts about ENat topology.